File: backend/src/main/java/spectacular/backend/specs/SpecLog.java
package spectacular.backend.specs;

import java.util.List;
import javax.validation.constraints.NotNull;

public class SpecLog {
  private final String id;
  private final SpecItem latestAgreed;
  private final List<ProposedSpecChange> proposedChanges;

  /**
   * Constructs a SpecLog object that represents the current state of a spec item's evolution.
   *
   * @param latestAgreed the current agreed version of the SpecItem
   * @param proposedChanges a list of the open ProposedSpecChange objects
   */
  public SpecLog(@NotNull SpecItem latestAgreed, @NotNull List<ProposedSpecChange> proposedChanges) {
    this.id = latestAgreed.getRepository().getNameWithOwner() + "/" + latestAgreed.getFilePath();
    this.latestAgreed = latestAgreed;
    this.proposedChanges = proposedChanges;
  }

  public String getId() {
    return id;
  }

  public SpecItem getLatestAgreed() {
    return latestAgreed;
  }

  public List<ProposedSpecChange> getProposedChanges() {
    return proposedChanges;
  }
}

File: web/src/__tests__/test-data-generator/spec-log.js
import Repository from './repository';
import SpecItem from './spec-item';

const generateSpecLog = ({
  filePath = 'specs/example-template.yaml',
  repository = Repository.generateRepository(),
  latestAgreed = SpecItem.generateSpecItem({ filePath, repository }),
} = {}) => {
  const id = `${repository.nameWithOwner}/${filePath}`;
  return {
    id,
    latestAgreed,
    proposedChanges: [],
  };
};

export default {
  generateSpecLog,
};
